src/main/java/org/openstreetmap/atlas/checks/validation/intersections/LineCrossingWaterBodyCheck.java

```diff
@@ -299,16 +301,19 @@ public boolean validCheckForObject(final AtlasObject object)
         // each edge will be marked explicitly.
         for (final AtlasItem crossingItem : invalidCrossingItems)
         {
+
             // Update the flag
             newFlag.addObject(crossingItem);
             if (!recordedOsmIds.contains(crossingItem.getOsmIdentifier()))
             {
+                final Set<Location> intersectingLocations = this.collectIntersectingLocations(atlas,
```

Comment:
This is inside the for loop for each crossing item, and it does not take into consideration the current crossing item. So would this not potentially produce duplicate points for each osm id in `invalidCrossingItems`?

Reply:
Oh, I missed that when I reviewed. Good point. I thought that crossingItem was being passed into this function and the function just produced points where this crossingItem and waterbody intersect.